s3/replication/common/experiments/kafka_consumer.py

```diff
@@ -0,0 +1,158 @@
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Kafka consumer code
+# Please note : Few changes are required in the replication
+# manager module to add this consumer code.
+# For more details, please refer the feature branch code:
+# https://github.com/priyanka25081999/cortx-multisite/tree/ps/kafka
+
+import json
+from ast import literal_eval
+from s3replicationmanager.prepare_job import PrepareReplicationJob
+# Make sure to install aiokafka module using the command
+# >> pip3 install aiokafka
+# aiokafka version used for testing : aiokafka==0.7.2
+from aiokafka import AIOKafkaConsumer
+from aiokafka.structs import OffsetAndMetadata, TopicPartition
+
+class KafkaMain:
+    consumer = None
+
+    '''
+    Job id to offset mapping
+      {
+        "job_id": { <topic> : <topic_name>,
+                    <partition> : <partition_name>
+                    <offset> : <offset_value>
+                  }
+            ...
+       }
+    '''
+    job_id_to_offset = {}
+    last_commited_offset = 0
+
+    def __init__(self):
+        """Initialize."""
+        # List store topic, partition and offset of each job
+        self._consumer_details = {}
+        self._job_id = 0
+
+    # Initialize the kafka consumer
+    # This method should be called from app.py class on startup.
+    def initialize(self, app):
+        self._app = app
+        KafkaMain.consumer = AIOKafkaConsumer(
+            'FDMI',
+            bootstrap_servers='127.0.0.1:9092',
+            group_id="test",
+            auto_offset_reset='latest',
+            enable_auto_commit=False)
+
+    # Getter methods
+    def get_topic(self, job_id):
+        """Returns the topic."""
+        return KafkaMain.job_id_to_offset[job_id]["Topic"]
+
+    def get_partition(self, job_id):
+        return KafkaMain.job_id_to_offset[job_id]["Partition"]
+
+    def get_offset(self, job_id):
+        return KafkaMain.job_id_to_offset[job_id]["Offset"]
+
+    # Remove job from job-id to offset mapping after commit
+    def remove_job_from_mapping(self, job_id):
+        removed_key = KafkaMain.job_id_to_offset.pop(job_id, None)
+        if removed_key is not None:
+            print("Removed job id from mapping : {}".format(removed_key))
+        else:
+            print("Job id [{}] does not exist".format(job_id))
+
+    # Send job for replication after consuming from kafka
+    async def sendJob(self, data):
+        print("Sending Jobs...")
+        fdmi_dict = literal_eval(data)
+        job_record = PrepareReplicationJob.from_fdmi(fdmi_dict)
+
+        if job_record is None:
+            print("BadRequest")
+            return
+
+        jobs_list = self._app['all_jobs']
+        if jobs_list.is_job_present(job_record["replication-id"]):
+            # Duplicate job.
+            print('Duplicate Job!')
+            return
+
+        job = jobs_list.add_job_using_json(job_record)
+        self._job_id = job.get_job_id()
```

Comment:
The `_job_id` can be defined as a local variable, not a class member.

Reply:
Limit the namespace/scope of a variable.
Only define and use them in necessary scope.

Reply:
sure. done